### PR TITLE
Remove allowSyntheticDefaultImports

### LINKED ISSUE #2352: TypeScript 编译 React 不过
### Version
2.1.6

### Environment
@type/react

### Reproduction link
[https://github.com/ant-design/ant-design-mobile/blob/master/components/tabs/index.tsx](https://github.com/ant-design/ant-design-mobile/blob/master/components/tabs/index.tsx)

### Steps to reproduce
```javascript
import { Tabs } from 'antd-mobile';
<Tabs>
```

### What is expected?
编译通过，参考 `antd`

[antd的代码](https://github.com/ant-design/ant-design/blob/master/components/tabs/index.tsx)

[antd-mobile的代码](https://github.com/ant-design/ant-design-mobile/blob/master/components/tabs/index.tsx)

### What is actually happening?
如果在代码里面把 `import React from 'react'` 换成和 `import * as React from 'react'` 就可以编译通过了。

---
antd-mobile 所有组件都是如此

### PR DESCRIPTION
**Pull request by qazyn951230**
Fix #2352 #2245

* [x] Make sure that you follow antd's [code convention](https://github.com/ant-design/ant-design/wiki/Code-convention-for-antd).
* [x] Run `npm run lint` and fix those errors before submitting in order to keep consistent code style.
* [x] Rebase before creating a PR to keep commit history clear.
* [x] Add some descriptions and refer relative issues for you PR.

Extra checklist:

**if** *isBugFix* **:**

  * [ ] Make sure that you add at least one unit test for the bug which you had fixed.

**elif** *isNewFeature* **:**

  * [ ] Update API docs for the component.
  * [ ] Update/Add demo to demonstrate new feature.
  * [ ] Update TypeScript definition for the component.
  * [ ] Add unit tests for the feature.

---
This change is [Reviewable](https://reviewable.io/reviews/ant-design/ant-design-mobile/2507)

### CODE REVIEW

File: tsconfig.test.json [hunk 1]
```diff
@@ -0,0 +1,6 @@
+{
+  "extends": "./tsconfig.json",
+  "compilerOptions": {
+    "allowSyntheticDefaultImports": true
```
Comment (doxiaodong) [inline tsconfig.test.json]: Remove this when test to make sure the components work right

Comment (qazyn951230) [reply tsconfig.test.json]: Fix tests by change `import * as ReactDOM from 'react-dom';` to `import ReactDOM from 'react-dom';` in JavaScript file. The `allowSyntheticDefaultImports` can't remove.

**Comment by warmhug** [comment]
改动有点大，估计会和很多其他 pr 冲突，等其他 pr 合并后再合并。
cc @yesmeck 这是否和 antd 的做法一致？

**Comment by paranoidjk** [comment]
参考这个改动再 review 下 https://github.com/ant-design/ant-design/pull/8218/files

**Comment by paranoidjk** [comment]
另外文档也改下，印象中以前有让 ts 用户开启 allowSyntheticDefaultImports

**Comment by doxiaodong** [comment]
这个会不会 影响 rn 呢

**Comment by yesmeck** [comment]
js 和 jsx 不需要改

**Comment by qazyn951230** [comment]
JavaScript files was restored.

**Comment by doxiaodong** [comment]
这个需要等 react-component 下的组件先改完才可以用吧，先不要合吧

**Comment by doxiaodong** [comment]
更新一下，这个会稍微延后一点，暂时还请 设置 `allowSyntheticDefaultImports: true` 使用

**Comment by warmhug** [comment]
RN 代码已经拆分并合并到 master 方便开发，需要解决此处冲突。

**Comment by doxiaodong** [comment]
因为最近 把 rn 拆分出去了导致冲突，需要先 rebase 解决冲突，然后开始跟进这个. @qazyn951230

**Comment by qazyn951230** [comment]
Codes updated.

**Comment by doxiaodong** [comment]
@warmhug  找谁看一下这个相关的 rmc-xxx 的 pr

**Comment by warmhug** [comment]
https://github.com/react-component/m-drawer/
listview ... 遗漏了不少。看 antd-mobile package json 依赖：
https://github.com/ant-design/ant-design-mobile/blob/dd23d03c2ea02c278864213324a93f43c3cbbe81/package.json#L42-L56

一个人统一改动，然后各 m-xx 组件 owner review 测试 Approve 即可

**Comment by doxiaodong** [comment]
> listview ... 遗漏了不少。看 antd-mobile package json 依赖：

漏掉的都是 js 写的 list-view drawer , 我对照 package.json 一个个改的

**Comment by doxiaodong** [comment]
@yesmeck  帮忙看一下 rmc 那些改动？

**Comment by galenjiang** [comment]
有进展吗？

**Comment by doxiaodong** [comment]
thank you , close for https://github.com/ant-design/ant-design-mobile/pull/3179

**Pull request closed without merging**